pkg/api/types.go

```diff
@@ -2263,3 +2263,83 @@ type ThirdPartyResourceData struct {
 
 	Data []byte `json:"name,omitempty"`
 }
+
+// PodSecurityPolicy governs the ability to make requests that affect the SecurityContext
+// that will be applied to a container.
+type PodSecurityPolicy struct {
+	TypeMeta   `json:",inline"`
+	ObjectMeta `json:"metadata,omitempty"`
+
+	// Privileged determines if a container can request to be run as privileged.
+	Privileged bool `json:"privileged,omitempty"`
+	// Capabilities is a list of capabilities that can be requested to add to the container.
+	Capabilities []Capability `json:"capabilities,omitempty"`
+	// HostPath determines if the policy allows containers to use the HostDir volume plugin
+	HostPath bool `json:"hostPath,omitempty"`
+	// HostNetwork determines if the policy allows the use of HostNetwork in the pod spec.
+	HostNetwork bool `json:"hostNetwork,omitempty"`
+	// HostPorts determines if the policy allows host ports in the containers.
+	HostPorts bool `json:"hostPorts,omitempty"`
+	// SELinuxContext is the strategy that will dictate what labels will be set in the SecurityContext.
+	SELinuxContext SELinuxContextStrategyOptions `json:"seLinuxContext,omitempty"`
+	// RunAsUser is the strategy that will dictate what RunAsUser is used in the SecurityContext.
+	RunAsUser RunAsUserStrategyOptions `json:"runAsUser,omitempty"`
+
+	// The users who have permissions to use this policy
+	Users []string `json:"users,omitempty"`
+	// The groups that have permission to use this policy
+	Groups []string `json:"groups,omitempty"`
```

Comment:
And yes, I think the connection between a user/group and a particular permission/role/policy/podsecuritypolicy/whatever is best expressed as a binding. When @deads2k @pweil- and I were discussing how this interacted with rolds/rolebindings in OpenShift, I was hesitant to introduce two different binding objects. I'm wondering if a more generic user/group binding could use an object reference to accomplish both role bindings and podsecuritypolicy bindings. I'd need to think about that more. @deads2k do you have thoughts on unifying things bound to a user/group?